temporal-sdk/src/main/java/io/temporal/internal/sync/POJOWorkflowImplementationFactory.java

```diff
@@ -349,22 +363,30 @@ public WorkflowOutput execute(WorkflowInput input) {
         }
       }
 
-      protected void newInstance() {
+      protected void newInstance(Optional<Payloads> input) {
         Func<?> factory = workflowInstanceFactories.get(workflowImplementationClass);
         if (factory != null) {
           workflow = factory.apply();
         } else {
           try {
-            workflow = workflowImplementationClass.getDeclaredConstructor().newInstance();
+            try {
+              workflow = workflowImplementationClass.getDeclaredConstructor().newInstance();
+            } catch (NoSuchMethodException e) {
```

Comment:
What if the user in their constructor makes a reflection call that has a no-such-method? I know it's not likely, but in general I would recommend a more explicit approach instead of exceptions for control flow here.

Reply:
That would not throw a `NoSuchMethodException` that would throw an `InvocationException` with cause `NoSuchMethodException`

Reply:
When I said "makes a reflection call" I didn't mean necessarily "makes a reflection method invocation". They could call `getDeclaredConstructor` just like us (or their serialization library might or whatever). The general statement of not using reflection exceptions for control flow here I think still applies.

If you _must_, I think you should call `newInstance()` outside of this try/catch, but I mention in another comment, I think validating the constructor to use should be done at registration time.

Reply:
Where are they calling `getDeclaredConstructor`?

Reply:
Oh, my mistake, I see what you're saying, yes this exception can only happen if the declared constructor does not exist. I still think we should pre-validate the constructor's presence/visibility/params/etc at registration time. Can close this thread in favor of the other one if you want.